httpclient/config_test.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright © 2024 Acronis International GmbH.
+
+Released under MIT license.
+*/
+
+package httpclient
+
+import (
+	"bytes"
+	"github.com/acronis/go-appkit/config"
+	"github.com/stretchr/testify/require"
+	"testing"
+	"time"
+)
+
+func TestConfigWithLoader(t *testing.T) {
+	yamlData := []byte(`
+retries:
+  max_retry_attempts: 30
```

Comment:
Configuration parameters should follow camelCase naming conventions to maintain consistency with other packages in go-appkit. Let’s rename them accordingly.

Additionally, the current implementation doesn’t allow enabling the retry mechanism without also enabling rate limiting. I suggest introducing an enabled: true|false flag for both features to provide more flexibility.